src/python-frontend/json_utils.h

```diff
@@ -94,6 +96,39 @@ JsonType &find_function(JsonType &json, const std::string &func_name)
   throw std::runtime_error("Function " + func_name + " not found\n");
 }
 
+template <typename JsonType>
+JsonType &find_imported_function(JsonType &ast, const std::string &func_name)
+{
+  for (auto &node : ast["body"])
+  {
+    if (node["_type"] == "ImportFrom" || node["_type"] == "Import")
+    {
+      for (const auto &name : node["names"])
+        if (name["name"] == func_name)
+          return node;
+    }
+  }
+  throw std::runtime_error("Function " + func_name + " not found\n");
+}
+
+template <typename JsonType>
+const std::string
+get_object_alias(const JsonType &ast, const std::string &obj_name)
+{
+  for (auto &node : ast["body"])
```

Comment:
This loop iterates through all nodes and names, which can become expensive for large ASTs. 

Could we consider stopping early once the target is found or preprocessing the AST into a more query-efficient structure?

Reply:
Thanks for the suggestion. The function stops early with a return once the target alias is found.